iterator/cdc.go

```diff
@@ -285,35 +316,29 @@ func (c *CDCIterator) getRecMap(item map[string]stypes.AttributeValue) map[strin
 	return stringMap
 }
 
-func (c *CDCIterator) getOpenCDCRec(rec stypes.Record) (opencdc.Record, error) {
+func (c *CDCIterator) getOpenCDCRec(recWithTS recordWithTimestamp) (opencdc.Record, error) {
+	rec := recWithTS.Record
+
 	newImage := c.getRecMap(rec.Dynamodb.NewImage)
 	oldImage := c.getRecMap(rec.Dynamodb.OldImage)
 	image := newImage
-	// use the old image to get the deleted record's keys
 	if rec.EventName == stypes.OperationTypeRemove {
 		image = oldImage
 	}
-	// prepare key and position
 	structuredKey := opencdc.StructuredData{
 		c.partitionKey: image[c.partitionKey],
 	}
 	if c.sortKey != "" {
-		structuredKey = opencdc.StructuredData{
-			c.partitionKey: image[c.partitionKey],
-			c.sortKey:      image[c.sortKey],
-		}
+		structuredKey[c.sortKey] = image[c.sortKey]
 	}
-	pos := position.Position{
-		IteratorType:   position.TypeCDC,
-		SequenceNumber: *rec.Dynamodb.SequenceNumber,
-		Time:           time.Now(),
-	}
-	cdcPos, err := pos.ToRecordPosition()
+
+	// lock position
```

Comment:
We should use a lock here.^

Reply:
do you mean a lock for each write or read on the position?